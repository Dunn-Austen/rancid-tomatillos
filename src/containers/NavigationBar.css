.header {
  align-items: center;
  background-color: #2D2D2F;
  display: flex;
  height: 8vh;
  justify-content: space-around;
}

.welcome-user {
  border-radius: 20px;
  color: #C654DE;
  font-family: 'Exo 2', sans-serif;
  font-size: 30px;
  font-weight: bold;
  margin: 0;
  text-decoration: underline;
  text-shadow: 1px 1px 3px black;
}

.login-button,
.logout-button,
.home-button  {
  border: 1px solid black;
  border-radius: 3px;
  box-shadow: 0 0 5px 2px white;
  font-size: 20px;
  height: 35px;
  line-height: 5px;
  width: 90px;
}

.login-button:hover  {
  background-color: #c654dd;;
  border: 1px groove black;
  color: white;
}

.logout-button:hover  {
  background-color: #c654dd;;
  border: 1px groove black;
  color: white;
}

.home-button:hover  {
  background-color: #c654dd;;
  border: 1px groove black;
  color: white;
}
