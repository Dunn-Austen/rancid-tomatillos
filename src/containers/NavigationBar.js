import React from 'react';
import { connect } from 'react-redux';

const NavigationBar = () => {
  return (
    <div>
      <h1>NavigationBar</h1>
    </div>
  );
};

export default connect(null, null)(NavigationBar);
