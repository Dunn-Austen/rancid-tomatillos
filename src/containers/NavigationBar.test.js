import React from 'react';
import { shallow } from 'enzyme';
import { NavigationBar } from './NavigationBar';

describe('NavigationBar', () => {
  let wrapper;

  it('should match the NavigationBar Snapshot when no one is logged in', () => {
    wrapper = shallow(<NavigationBar
        user={{}}
        isLoggedIn={false}
        updateUser={jest.fn()}
        updateLoggedInStatus={jest.fn()}
      />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when user is logged in', () => {
    wrapper = shallow(<NavigationBar
      user={{ id: 22, name: 'Marge', email: 'marge@turing.io', ranking: [] }}
      isLoggedIn={true}
      updateUser={jest.fn()}
      updateLoggedInStatus={jest.fn()}
      />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke updateUser props with empty object when log out button is clicked',
    () => {
      const mockUpdateUser = jest.fn();
      wrapper = shallow(<NavigationBar
        user={{ id: 22, name: 'Marge', email: 'marge@turing.io', ranking: [] }}
        isLoggedIn={true}
        updateUser={mockUpdateUser}
        updateLoggedInStatus={jest.fn()}
        />);

      wrapper.find('.logout-button').simulate('click');
      expect(mockUpdateUser).toHaveBeenCalledWith({});
  });

  it('should invoke updateLoggedInStatus props with false status when button is clicked',
    () => {
      const mockUpdateLoggedInStatus = jest.fn();
      wrapper = shallow(<NavigationBar
        user={{ id: 22, name: 'Marge', email: 'marge@turing.io', ranking: [] }}
        isLoggedIn={true}
        updateUser={jest.fn()}
        updateLoggedInStatus={mockUpdateLoggedInStatus}
        />);

      wrapper.find('.logout-button').simulate('click');
      expect(mockUpdateLoggedInStatus).toHaveBeenCalledWith(false);
  });

  // without Redux we would test that updateUser and updateLoggedInStatus update
  // state (which is the Redux store).  With Redux, do we still test this? how?
});
