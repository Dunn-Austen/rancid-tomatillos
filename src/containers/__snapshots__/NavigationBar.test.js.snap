// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar Container NavigationBar component should match snapshot when user is logged in 1`] = `
<header
  className="header"
>
  <Link
    to="/"
  >
    <Route
      path={
        Array [
          "/login",
          "/movies/:id",
        ]
      }
      render={[Function]}
    />
  </Link>
  <h1
    className="welcome-user"
  >
    Welcome 
    Marge
  </h1>
  <Link
    to="/"
  >
    <button
      className="logout-button"
      onClick={[Function]}
    >
      Log Out
    </button>
  </Link>
</header>
`;

exports[`NavigationBar Container NavigationBar component should match the NavigationBar Snapshot when no one is logged in 1`] = `
<header
  className="header"
>
  <Link
    to="/"
  >
    <Route
      path={
        Array [
          "/login",
          "/movies/:id",
        ]
      }
      render={[Function]}
    />
  </Link>
  <Link
    to="/login"
  >
    <button
      className="login-button"
    >
      Log In
    </button>
  </Link>
</header>
`;
